FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+  bitrate?: number;
+  duration?: number;
+  originalPath?: string;
+}
+// selected document format interface
+interface docDefaultFormat {
+  name: string;
+  size: number;
+  fileCopyUri: null;
+  type: string;
+  uri: string;
+}
+
+const CreatePost = () => {
+  const memberData = useAppSelector(state => state.feed.member);
+  const dispatch = useDispatch();
+  const [formattedDocAttachments, setFormattedDocAttachments] = useState<
+    Array<LMAttachmentUI>
+  >([]);
+  const [formattedMediaAttachments, setFormattedMediaAttachments] = useState<
+    Array<LMAttachmentUI>
+  >([]);
+  const [formattedLinkAttachments, setFormattedLinkAttachments] = useState<
+    Array<LMAttachmentUI>
+  >([]);
+  const [showOptions, setShowOptions] = useState(true);
+  const [showPreview, setShowPreview] = useState(false);
+  const [postContentText, setPostContentText] = useState('');
+  const MAX_FILE_SIZE = 104857600; // 100MB in bytes
+
+  // convert the default selected image/video format to LMAttachmentUI
+  function convertData(data: imageDefaultFormat[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: item.type,
+          name: item.fileName,
+          ogTags: {
+            description: '',
+            title: '',
+            url: '',
+            image: '',
+          },
+          size: item.fileSize,
+          duration: item.duration, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'image/jpg'
+            ? IMAGE_ATTACHMENT_TYPE
+            : item.type === 'video/mp4'
+            ? VIDEO_ATTACHMENT_TYPE
+            : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedMediaAttachments([
+      ...formattedMediaAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
+
+  // convert the default selected document format to LMAttachmentUI
+  function convertDocData(data: docDefaultFormat[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: item.type,
+          name: item.name,
+          ogTags: {
+            description: '',
+            title: '',
+            url: '',
+            image: '',
+          },
+          size: item.size,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'application/pdf' ? DOCUMENT_ATTACHMENT_TYPE : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedDocAttachments([...formattedDocAttachments, ...convertedData]);
+    return convertedData;
+  }
```

Comment:
move this convertor to a new file and then use returned data in state. and rename `convertDocumentMetaData`

Reply:
resolved